identity_base_config/recipes/default.rb

```diff
@@ -74,7 +74,8 @@
 # download Slack hook/channel files for id-rails-console
 %w(slackwebhook slackchannel).each do |f|
   file "/etc/login.gov/keys/#{f}" do
-    content ConfigLoader.load_config(node, f)
+    content ConfigLoader.load_config_or_nil(node, f)
+    not_if { node['config_loader']['bucket_prefix'].nil? }
```

Comment:
Why the change to `_or_nil()`? I'd think it should be an error if the webhook is missing.
I'm also not sure why the `not_if` block is needed.

Reply:
This is an issue that the AMI build has. When it gets to this block, the objects it attempts to load (via `load_config`) begin with `_default` instead of a URL, and since that doesn't exist (and isn't `nil`), it errors out and the Chef AMI build fails.

Reply:
Oh, I understand now. I wasn't really thinking about the context previously.

None of this delivery of secrets should be in this repo at all.

The base image delivers static files that don't differ between environments, such as the content of scripts like `id-rails-console.`

For delivering secrets like `slackwebhook`, that can't happen in the base image, because the values differ between environments, and are sensitive and shouldn't be baked into the AMI.

So this ConfigLoader.load_config() call should be happening in a recipe in identity-devops instead.

That's why it's especially good practice to separate out the static script content from the delivery of the secrets.

Reply:
(That also means that we don't need the config_loader or aws_metadata cookbooks in this repo, although it's not harmful to move them.)